pkg/resource/generator.go

```diff
@@ -285,5 +287,25 @@ func (g *Generator) Remove(cr *imageregistryv1.Config, modified *bool) error {
 		*modified = true
 	}
 
+	driver, err := storage.NewDriver(cr.ObjectMeta.Name, cr.ObjectMeta.Namespace, &cr.Status.Storage)
+	if err != nil {
+		return err
+	}
+
+	var derr error
+	err = wait.PollImmediate(1*time.Second, 5*time.Minute, func() (stop bool, err error) {
+		if retriable, derr := driver.RemoveStorage(cr, modified); derr != nil {
```

Comment:
this isn't going to preserve derr the way you want, you've created a shadow copy of it.
https://play.golang.org/p/hPuFAc7DVYl